src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -684,7 +684,7 @@ InterpreterSelectQuery::InterpreterSelectQuery(
         if (!options.only_analyze)
         {
             if (query.sampleSize() && (input_pipe || !storage || !storage->supportsSampling()))
-                throw Exception(ErrorCodes::SAMPLING_NOT_SUPPORTED, "Illegal SAMPLE: table doesn't support sampling");
+                throw Exception(ErrorCodes::SAMPLING_NOT_SUPPORTED, "Illegal SAMPLE: table {} doesn't support sampling", storage->getStorageID().getNameForLogs());
```

Comment:
storage can be `nullptr` here so SEGFAULT happens

Reply:
https://s3.amazonaws.com/clickhouse-test-reports/0/7b143926754c8450c70a98afb882414d24b2090d/fuzzer_astfuzzermsan/report.html